connector/signaltometricsconnector/testdata/logs/exponential_histograms/config.yaml

```diff
@@ -5,6 +5,14 @@ signaltometrics:
       exponential_histogram:
         count: "1"
         value: attributes["log.duration"]
+    - name: total.logrecords.resource.foo.exphistogram
+      description: Logrecords with resource attribute foo as exponential histogram with log.duration from attributes
+      collector_info_as_resource_attributes: true
+      include_resource_attributes:
```

Comment:
I'm wondering whether we should implicitly add the collector info if `include_resource_attributes` is used. Maybe with an opt-out. But by default, it seems that using `include_resource_attributes`  always bears the risk of violating single-writer.

Reply:
I deliberated on that, and I chose the current approach because it seemed a bit less confusing and magical. However, I do agree that include_resource_attributes has a high risk of violating single-writer.

To avoid confusion, we can do something like, always (include_resource_attributes present or not) add the collector info as resource attribute and give an opt-out option to the users?

Reply:
> IIANM even if you included all resource attributes (i.e. include_resource_attributes is left unconfigured) then you could still violate the single-writer principle.

Why is that? Is it because signals other than metrics don't require to have a single-writer, so if you create metrics from these signals, you can't assume a single writer either?

Reply:
As discussed live, yes, because single-writer is only relevant for metrics, and a log record may have non-unique resource attributes. @lahsivjar please correct me if I'm wrong.

Reply:
++ unless @felixbarny has objections let's remove the config and always add it

Reply:
sgtm

Reply:
Updated to always add the collector info as resource attribute.